src/tasks/pair-liquidity-info-history-importer.service.ts

```diff
@@ -0,0 +1,189 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { MdwClientService } from '../clients/mdw-client.service';
+import { PairService, PairWithTokens } from '../database/pair.service';
+import { isEqual, uniqWith, values } from 'lodash';
+import { PairLiquidityInfoHistoryService } from '../database/pair-liquidity-info-history.service';
+import { contractIdToAccountId } from '../lib/utils';
+import { PairLiquidityInfoHistoryErrorService } from '../database/pair-liquidity-info-history-error.service';
+import { getClient } from '../lib/contracts';
+import { Cron, CronExpression } from '@nestjs/schedule';
+
+type MicroBlock = {
+  hash: string;
+  timestamp: bigint;
+  height: number;
+};
+
+@Injectable()
+export class PairLiquidityInfoHistoryImporterService {
+  constructor(
+    private mdwClientService: MdwClientService,
+    private pairService: PairService,
+    private pairLiquidityInfoHistoryService: PairLiquidityInfoHistoryService,
+    private pairLiquidityInfoHistoryErrorService: PairLiquidityInfoHistoryErrorService,
+  ) {}
+
+  private readonly logger = new Logger(
+    PairLiquidityInfoHistoryImporterService.name,
+  );
+
+  private isSyncRunning: boolean = false;
+
+  // TODO change to desired frequency
+  @Cron(CronExpression.EVERY_30_SECONDS)
+  async runTask() {
+    try {
+      if (!this.isSyncRunning) {
+        this.isSyncRunning = true;
+        await this.syncPairLiquidityInfoHistory();
+        this.isSyncRunning = false;
+      }
+    } catch (error) {
+      this.isSyncRunning = false;
+      this.logger.error(`Sync failed. ${error}`);
+    }
+  }
+
+  private async syncPairLiquidityInfoHistory() {
+    this.logger.log(`Started syncing pair liquidity info history.`);
+
+    // Fetch all pairs from DB
+    const pairsWithTokens = await this.pairService.getAll();
+    this.logger.log(
+      `Syncing liquidity info history for ${pairsWithTokens.length} pairs.`,
+    );
+
+    for (const pairWithTokens of pairsWithTokens) {
+      try {
+        // Get current height
+        const currentHeight = await (await getClient())[0].getHeight();
```

Comment:
Not a huge fan of double await
```suggestion
        const currentHeight = await getClient().then(([client]) => client.getHeight());
```